bson/codec_options.py

```diff
@@ -294,27 +286,35 @@ class CodecOptions(_options_base):
 
     def __new__(
         cls: Type["CodecOptions"],
-        document_class: _DocumentClass = dict,
+        document_class: Optional[Type[Mapping[str, Any]]] = None,
         tz_aware: bool = False,
         uuid_representation: Optional[int] = UuidRepresentation.UNSPECIFIED,
-        unicode_decode_error_handler: Optional[str] = "strict",
+        unicode_decode_error_handler: str = "strict",
         tzinfo: Optional[datetime.tzinfo] = None,
         type_registry: Optional[TypeRegistry] = None,
     ) -> "CodecOptions":
-        if not (issubclass(document_class, _MutableMapping) or _raw_document_class(document_class)):
+        doc_class = document_class or dict
+        # issubclass can raise TypeError for generic aliases like SON[str, Any].
+        # In that case we can use the base class for the comparison.
+        is_mapping = False
+        try:
+            is_mapping = issubclass(doc_class, _MutableMapping)
+        except TypeError:
+            is_mapping = issubclass(doc_class.__origin__, _MutableMapping)
```

Comment:
This should  protect against the case where doc_class.__origin__ does not exist:
```
getattr(doc_class, "__origin__", type)
```

Reply:
Already fixed.  ;)